docs/guides/hooks.md

```diff
@@ -0,0 +1,85 @@
+# Hooks
+Hooks exist so that users can latch on to certain Video.js lifecycle events
+
+
+## Current Hooks
+Currently the following lifecycle events are accept and use hooks
+
+### beforesetup
+Called just before the player is created, so that the user can modify the options passed to the player or modify the video element that will be used for the player
```

Comment:
I think we should mention here (and not just in the example) what happens if multiple hooks are.
Also, state that the method takes in the video element and an options object and is expected to return a new options object.